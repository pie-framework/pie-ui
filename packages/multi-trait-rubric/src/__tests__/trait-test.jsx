import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Trait from '../trait';

jest.mock('@pie-lib/render-ui', () => ({
  color: {
    primaryDark: jest.fn().mockReturnValue('blue'),
    secondaryBackground: jest.fn().mockReturnValue('grey'),
  }
}));

describe('Trait', () => {
  let wrapper, trait, options;

  const mkWrapper = (traitOptions = {}, showOptions = {}) => {
    trait = {
      name: 'Trait 1',
      description: 'Trait 1 Description',
      scorePointsDescriptors: ['Descriptor 0', 'Descriptor 1'],
      standards: ['a', 'b'],
      ...traitOptions
    };

    options = {
      scorePointsValues: [0, 1],
      showStandards: true,
      showDescription: true,
      ...showOptions
    };

    return mount(<Trait
      trait={trait}
      traitIndex={1}
      scaleIndex={1}
      {...options}
    />);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    it('renders without standards, description and scorePointsValues', () => {
      const wrapper = mkWrapper({}, { showStandards: false, showDescription: false, scorePointsValues: [] });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders without scorePointsDescriptors', () => {
      const wrapper = mkWrapper({ scorePointsDescriptors: [] }, {});
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
