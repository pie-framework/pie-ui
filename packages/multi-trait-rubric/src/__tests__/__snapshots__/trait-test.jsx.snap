// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trait snapshots renders without scorePointsDescriptors 1`] = `
<WithStyles(Trait)
  scaleIndex={1}
  scorePointsValues={
    Array [
      0,
      1,
    ]
  }
  showDescription={true}
  showStandards={true}
  trait={
    Object {
      "description": "Trait 1 Description",
      "name": "Trait 1",
      "scorePointsDescriptors": Array [],
      "standards": Array [
        "a",
        "b",
      ],
    }
  }
  traitIndex={1}
>
  <Trait
    classes={
      Object {
        "trait": "Trait-trait-1",
        "traitName": "Trait-traitName-2",
      }
    }
    scaleIndex={1}
    scorePointsValues={
      Array [
        0,
        1,
      ]
    }
    showDescription={true}
    showStandards={true}
    trait={
      Object {
        "description": "Trait 1 Description",
        "name": "Trait 1",
        "scorePointsDescriptors": Array [],
        "standards": Array [
          "a",
          "b",
        ],
      }
    }
    traitIndex={1}
  >
    <tr
      key="scale-1-trait-1"
    >
      <td
        className="Trait-trait-1 Trait-traitName-2"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Trait 1",
            }
          }
        />
      </td>
      <td>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "a,b",
            }
          }
        />
      </td>
      <td>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Trait 1 Description",
            }
          }
        />
      </td>
      <td
        key="table-cell-0"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
      <td
        key="table-cell-1"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </td>
    </tr>
  </Trait>
</WithStyles(Trait)>
`;

exports[`Trait snapshots renders without standards, description and scorePointsValues 1`] = `
<WithStyles(Trait)
  scaleIndex={1}
  scorePointsValues={Array []}
  showDescription={false}
  showStandards={false}
  trait={
    Object {
      "description": "Trait 1 Description",
      "name": "Trait 1",
      "scorePointsDescriptors": Array [
        "Descriptor 0",
        "Descriptor 1",
      ],
      "standards": Array [
        "a",
        "b",
      ],
    }
  }
  traitIndex={1}
>
  <Trait
    classes={
      Object {
        "trait": "Trait-trait-1",
        "traitName": "Trait-traitName-2",
      }
    }
    scaleIndex={1}
    scorePointsValues={Array []}
    showDescription={false}
    showStandards={false}
    trait={
      Object {
        "description": "Trait 1 Description",
        "name": "Trait 1",
        "scorePointsDescriptors": Array [
          "Descriptor 0",
          "Descriptor 1",
        ],
        "standards": Array [
          "a",
          "b",
        ],
      }
    }
    traitIndex={1}
  >
    <tr
      key="scale-1-trait-1"
    >
      <td
        className="Trait-trait-1 Trait-traitName-2"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Trait 1",
            }
          }
        />
      </td>
    </tr>
  </Trait>
</WithStyles(Trait)>
`;
